<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>355</width>
    <height>594</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <widget class="QTableWidget" name="myTable">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>10</y>
      <width>151</width>
      <height>571</height>
     </rect>
    </property>
    <column>
     <property name="text">
      <string>Массив</string>
     </property>
    </column>
   </widget>
   <widget class="QComboBox" name="arrayTypeBox">
    <property name="geometry">
     <rect>
      <x>170</x>
      <y>10</y>
      <width>171</width>
      <height>31</height>
     </rect>
    </property>
    <item>
     <property name="text">
      <string>Целочисленный массив</string>
     </property>
    </item>
    <item>
     <property name="text">
      <string>Вешественный массив</string>
     </property>
    </item>
   </widget>
   <widget class="QSpinBox" name="spinBox">
    <property name="geometry">
     <rect>
      <x>170</x>
      <y>150</y>
      <width>61</width>
      <height>25</height>
     </rect>
    </property>
    <property name="minimum">
     <number>-100</number>
    </property>
    <property name="maximum">
     <number>100</number>
    </property>
   </widget>
   <widget class="QDoubleSpinBox" name="doubleSpinBox">
    <property name="geometry">
     <rect>
      <x>250</x>
      <y>150</y>
      <width>91</width>
      <height>25</height>
     </rect>
    </property>
    <property name="decimals">
     <number>3</number>
    </property>
    <property name="minimum">
     <double>-100.000000000000000</double>
    </property>
    <property name="maximum">
     <double>100.000000000000000</double>
    </property>
    <property name="singleStep">
     <double>0.001000000000000</double>
    </property>
   </widget>
   <widget class="QPushButton" name="addButton">
    <property name="geometry">
     <rect>
      <x>170</x>
      <y>190</y>
      <width>171</width>
      <height>31</height>
     </rect>
    </property>
    <property name="text">
     <string>Добавить элемент</string>
    </property>
   </widget>
   <widget class="QPushButton" name="delButton">
    <property name="geometry">
     <rect>
      <x>170</x>
      <y>240</y>
      <width>171</width>
      <height>31</height>
     </rect>
    </property>
    <property name="text">
     <string>Удалить элемент</string>
    </property>
   </widget>
   <widget class="QComboBox" name="searchTypeBox">
    <property name="geometry">
     <rect>
      <x>170</x>
      <y>290</y>
      <width>171</width>
      <height>31</height>
     </rect>
    </property>
    <item>
     <property name="text">
      <string>Линейный поиск</string>
     </property>
    </item>
    <item>
     <property name="text">
      <string>Бинарный поиск</string>
     </property>
    </item>
    <item>
     <property name="text">
      <string>Бинарный поиск первое и последнее</string>
     </property>
    </item>
    <item>
     <property name="text">
      <string>Бинарный поиск все</string>
     </property>
    </item>
   </widget>
   <widget class="QPushButton" name="searchButton">
    <property name="geometry">
     <rect>
      <x>170</x>
      <y>380</y>
      <width>171</width>
      <height>31</height>
     </rect>
    </property>
    <property name="text">
     <string>Найти</string>
    </property>
   </widget>
   <widget class="QPushButton" name="clearButton">
    <property name="geometry">
     <rect>
      <x>170</x>
      <y>430</y>
      <width>171</width>
      <height>31</height>
     </rect>
    </property>
    <property name="text">
     <string>Очистить таблицу и массив</string>
    </property>
   </widget>
   <widget class="QDoubleSpinBox" name="searchDoubleSpinBox">
    <property name="geometry">
     <rect>
      <x>250</x>
      <y>340</y>
      <width>91</width>
      <height>25</height>
     </rect>
    </property>
    <property name="decimals">
     <number>3</number>
    </property>
    <property name="minimum">
     <double>-100.000000000000000</double>
    </property>
    <property name="maximum">
     <double>100.000000000000000</double>
    </property>
    <property name="singleStep">
     <double>0.001000000000000</double>
    </property>
   </widget>
   <widget class="QSpinBox" name="searchSpinBox">
    <property name="geometry">
     <rect>
      <x>170</x>
      <y>340</y>
      <width>61</width>
      <height>25</height>
     </rect>
    </property>
    <property name="minimum">
     <number>-100</number>
    </property>
    <property name="maximum">
     <number>100</number>
    </property>
   </widget>
   <widget class="QPushButton" name="addButton_2">
    <property name="geometry">
     <rect>
      <x>170</x>
      <y>100</y>
      <width>171</width>
      <height>31</height>
     </rect>
    </property>
    <property name="text">
     <string>Выделить память</string>
    </property>
   </widget>
   <widget class="QLabel" name="label">
    <property name="geometry">
     <rect>
      <x>170</x>
      <y>60</y>
      <width>121</width>
      <height>21</height>
     </rect>
    </property>
    <property name="text">
     <string>Кол - во элементов</string>
    </property>
   </widget>
   <widget class="QSpinBox" name="sizeSpinBox">
    <property name="geometry">
     <rect>
      <x>290</x>
      <y>60</y>
      <width>51</width>
      <height>25</height>
     </rect>
    </property>
    <property name="minimum">
     <number>1</number>
    </property>
    <property name="maximum">
     <number>100</number>
    </property>
    <property name="value">
     <number>1</number>
    </property>
   </widget>
  </widget>
 </widget>
 <resources/>
 <connections/>
</ui>
